test/unit/save_state.js

```diff
@@ -0,0 +1,25 @@
+var fs = require('fs');
+var path = require('path');
+var test = require('tap').test;
+var VirtualMachine = require('../../src/index');
+
+var uri = path.resolve(__dirname, '../fixtures/default.json');
+var file = fs.readFileSync(uri, 'utf8');
+
+test('save state', function (t) {
+    var vm = new VirtualMachine();
+
+    // Start VM, load project, and run
+    t.doesNotThrow(function () {
+        vm.start();
+        vm.clear();
+        vm.setCompatibilityMode(false);
+        vm.setTurboMode(false);
+        vm.loadProject(file);
+        var value = vm.testJSON();
+    });
+
+    t.ok(value);
```

Comment:
This is sufficient for an integration test, but for unit testing we expect more coverage. I would recommend running `npm run coverage` and modifying your tests until you have reached 100% coverage for your additions.